StripeCore/StripeCore/Source/DownloadManager/DownloadManager.swift

```diff
@@ -125,18 +140,34 @@ extension DownloadManager {
             return image
         }
         let urlRequest = URLRequest(url: url, cachePolicy: .useProtocolCachePolicy)
-        let task = self.session.downloadTask(with: url) { [weak self] tempURL, response, _ in
-            guard let self = self,
-                  let tempURL = tempURL,
-                let response = response,
-                let data = self.getDataFromURL(tempURL),
-                let image = self.persistToMemory(data, forImageName: imageName)
-            else {
-                self?.pendingRequestsSemaphore.wait()
-                self?.pendingRequests.removeValue(forKey: imageName)
-                self?.pendingRequestsSemaphore.signal()
-                let errorAnalytic = ErrorAnalytic(event: .unexpectedStripeCoreDownloadManagerError,
-                                                  error: Error.downloadAsyncFailure,
+        let task = self.session.downloadTask(with: url) { [weak self] tempURL, response, responseError in
```

Comment:
It feels like we could (not in this PR) rewrite this class to implement this method to simply call the other one or vice versa.